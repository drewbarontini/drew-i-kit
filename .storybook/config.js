import { configure, addDecorator } from '@storybook/react';
import React, { Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import 'normalize.css';

import { theme } from '../config';

import GlobalStyles from '../components/GlobalStyles';

const req = require.context('../components/', true, /.story.js$/);
function loadStories() {
  req.keys().forEach(filename => req(filename));
}

function withStyledComponents(story) {
  return (
    <ThemeProvider theme={theme}>
      <Fragment>
        {story()}
        <GlobalStyles />
      </Fragment>
    </ThemeProvider>
  );
}

addDecorator(withStyledComponents);
configure(loadStories, module);
