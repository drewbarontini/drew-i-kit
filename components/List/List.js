import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

import { theme, setProp } from '../../lib/utils';

const List = styled.ul`
  margin: 0;
  padding: 0;

  ${setProp({
    prop: 'type',
    cssProp: 'list-style-type',
    fallback: 'none',
  })}

  ${ifProp(
    'type',
    css`
      margin-left: ${theme('spacing.base')};
    `
  )}

  > *:not(:last-child) {
    margin-bottom: ${props => (props.flush ? '0' : theme('spacing.s'))};
  }
`;

List.propTypes = {
  flush: PropTypes.string,
  type: PropTypes.string,
};

List.displayName = 'List';

export default List;
