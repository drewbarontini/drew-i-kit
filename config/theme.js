const theme = {
  borders: {
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  spacing: {
    base: '16px',
    xs: '5px',
    s: '10px',
    m: '12px',
    l: '32px',
    xl: '60px',
  },
};

export default theme;
