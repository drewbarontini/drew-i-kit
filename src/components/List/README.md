# List

### Usage

```jsx
import { List } from 'drew-i-kit';

<List>
  <li>One</li>
  <li>Two</li>
  <li>Three</li>
  <li>Four</li>
</List>
```

```jsx
import { List } from 'drew-i-kit';

<List type="disc">
  <li>One</li>
  <li>Two</li>
  <li>Three</li>
  <li>Four</li>
</List>
<List type="decimal">
  <li>One</li>
  <li>Two</li>
  <li>Three</li>
  <li>Four</li>
</List>
```

### Props

- `flush` (**Boolean**, optional)
- `type` (**string**, optional)