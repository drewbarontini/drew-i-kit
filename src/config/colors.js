import { paletteFor } from '../lib';

const colors = {
  ...paletteFor({
    area: 'primary',
    base: '#3576a7',
    darker: ['#152f43', '#1d415c', '#255375', '#2d648e'],
    lighter: ['#538bb4', '#729fc1', '#90B4cf', '#aeC8dc'],
  }),
  ...paletteFor({
    area: 'neutral',
    base: '#737373',
    darker: ['#0b0b0b', '#1d1d1d', '#2e2e2e', '#3f3f3f', '#515151', '#626262'],
    lighter: ['#888888', '#9d9d9d', '#b2b2b2', '#c7c7c7', '#dcdcdc', '#f1f1f1'],
  }),
  accent: {
    ...paletteFor({
      area: 'green',
      base: '#8acb88',
      darker: ['#233322', '#375136', '#4c704b', '#618e5f', '#75ad74'],
      lighter: ['#9cd39a', '#addbac', '#bfe2be', '#d0eacf', '#e2f2e1'],
    }),
    ...paletteFor({
      area: 'yellow',
      base: '#ffc527',
      darker: ['#40310a', '#664f10', '#8c6c15', '#b38a1b', '#d9a721'],
      lighter: ['#ffce47', '#ffd668', '#ffdf88', '#ffe8a9', '#fff1c9'],
    }),
    ...paletteFor({
      area: 'red',
      base: '#fe5f55',
      darker: ['#401815', '#662622', '#8c342f', '#b2433b', '#d85148'],
      lighter: ['#fe776f', '#fe8f88', '#fea7a2', '#ffbfbb', '#ffd7d5'],
    }),
  },
  bg: '#fff',
  fg: '#272727',
  white: '#fff',
  black: '#272727',
  border: '#dcdcdc',
};

export default colors;
