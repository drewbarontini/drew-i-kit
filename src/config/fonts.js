const fonts = {
  family: {
    base:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    heading:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    monospace: "'Monaco', monospace",
  },
  lineHeight: {
    base: '1.5',
    heading: '1.2',
  },
  sizes: {
    base: '14px',
    xxs: '8px',
    xs: '10px',
    s: '12px',
    l: '16px',
    xl: '24px',
    h1: '30px',
    h2: '26px',
    h3: '22px',
    h4: '18px',
  },
  weights: {
    light: 200,
    normal: 400,
    semiBold: 600,
    bold: 800,
  },
};

export default fonts;
