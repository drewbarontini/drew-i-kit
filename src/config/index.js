export { default as cssProps } from './cssProps';
export { default as icons } from './icons';
export { default as theme } from './theme';
