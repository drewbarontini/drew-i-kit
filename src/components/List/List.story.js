import React from 'react';
import { storiesOf } from '@storybook/react';

import List from './';

storiesOf('List', module)
  .add('default', () => (
    <List>
      <li>One</li>
      <li>Two</li>
      <li>Three</li>
      <li>Four</li>
    </List>
  ))
  .add('styled', () => (
    <List type="disc">
      <li>One</li>
      <li>Two</li>
      <li>Three</li>
      <li>Four</li>
    </List>
  ))
  .add('numbered', () => (
    <List type="decimal">
      <li>One</li>
      <li>Two</li>
      <li>Three</li>
      <li>Four</li>
    </List>
  ));
