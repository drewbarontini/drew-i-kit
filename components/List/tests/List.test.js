import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import { theme } from '../../../config';

import List from '../';

describe('<List />', () => {
  it('should render properly', () => {
    const { container, queryAllByTestId } = render(
      <List theme={theme}>
        <li data-testid="li">One</li>
        <li data-testid="li">Two</li>
        <li data-testid="li">Three</li>
      </List>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(queryAllByTestId('li')).toHaveLength(3);
  });
});
