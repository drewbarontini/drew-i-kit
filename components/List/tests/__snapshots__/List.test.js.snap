// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render properly 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c0 > *:not(:last-child) {
  margin-bottom: 10px;
}

<ul
  class="c0"
>
  <li
    data-testid="li"
  >
    One
  </li>
  <li
    data-testid="li"
  >
    Two
  </li>
  <li
    data-testid="li"
  >
    Three
  </li>
</ul>
`;
